test/unit/lib/plugins/aws/package/compile/events/kafka.test.js

```diff
@@ -0,0 +1,534 @@
+        cfTemplate.Resources[naming.getKafkaEventLogicalId('other', 'TestingTopic')];
+      defaultIamRole = cfTemplate.Resources.IamRoleLambdaExecution;
+    });
+
+    it('should correctly compile EventSourceMapping resource properties with minimal configuration', () => {
+      expect(minimalEventSourceMappingResource.Properties).to.deep.equal({
+        SelfManagedEventSource: {
+          Endpoints: {
+            KafkaBootstrapServers: ['abc.xyz:9092'],
+          },
+        },
+        SourceAccessConfigurations: [
+          {
+            Type: 'SASL_SCRAM_256_AUTH',
+            URI: saslScram256AuthArn,
+          },
+        ],
+        StartingPosition: 'TRIM_HORIZON',
+        Topics: [topic],
+        FunctionName: {
+          'Fn::GetAtt': [naming.getLambdaLogicalId('foo'), 'Arn'],
+        },
+      });
+    });
+
+    it('should update default IAM role with SecretsManager statement', () => {
+      expect(defaultIamRole.Properties.Policies[0].PolicyDocument.Statement).to.deep.include({
+        Effect: 'Allow',
+        Action: ['secretsmanager:GetSecretValue'],
+        Resource: [saslScram256AuthArn],
+      });
+    });
+
+    it('should correctly compile EventSourceMapping resource DependsOn ', () => {
+      expect(minimalEventSourceMappingResource.DependsOn).to.equal('IamRoleLambdaExecution');
+      expect(allParamsEventSourceMappingResource.DependsOn).to.equal('IamRoleLambdaExecution');
+    });
+
+    it('should correctly compile EventSourceMapping resource with all parameters', () => {
+      expect(allParamsEventSourceMappingResource.Properties).to.deep.equal({
+        BatchSize: batchSize,
+        Enabled: enabled,
+        SelfManagedEventSource: {
+          Endpoints: {
+            KafkaBootstrapServers: ['abc.xyz:9092'],
+          },
+        },
+        SourceAccessConfigurations: [
+          {
+            Type: 'SASL_SCRAM_256_AUTH',
+            URI: saslScram256AuthArn,
+          },
+        ],
+        StartingPosition: startingPosition,
+        Topics: [topic],
+        FunctionName: {
+          'Fn::GetAtt': [naming.getLambdaLogicalId('other'), 'Arn'],
+        },
+      });
+    });
+  });
+
+  describe('configuring kafka events', () => {
+    const runCompileEventSourceMappingTest = async (eventConfig) => {
+      const { awsNaming, cfTemplate } = await runServerless({
+        fixture: 'function',
+        configExt: {
+          functions: {
+            foo: {
+              events: [
+                {
+                  kafka: eventConfig.event,
+                },
+              ],
+            },
+          },
+        },
+        cliArgs: ['package'],
+      });
+
+      const eventSourceMappingResource =
+        cfTemplate.Resources[awsNaming.getKafkaEventLogicalId('foo', 'TestingTopic')];
+      expect(eventSourceMappingResource.Properties).to.deep.equal(eventConfig.resource(awsNaming));
+    };
+
+    const requirePropertyTest = async (event) => {
```

Comment:
Our standard approach is not to test schema validation in such manner - we instead test all supported scenarios which you already have. I believe these tests could be removed